### ISSUE TITLE
Error when running bin/magento setup:di:compile when installing on 2.3.6 and up

### ISSUE THREAD

**Issue by duckchip** [distrimedia/magento2-distrimedia-connector#1]
### Description

After upgrading from 2.3.5-p1 to 2.3.6 an undocumented breaking change was introduced to the \Magento\Sales\Model\Order\Pdf\Invoice class. This prevents invoices from being printed if the class has been extended.

### Preconditions:
connector v1.0.1
magento 2.3.6 and up 

### Steps to reproduce
Install the module using composer and run `bin/magento setup:di:compile`

### Expected result
Succesful generation of the interceptors

### Actual result
```
Interception cache generation... 6/9 [==================>---------]  66% 42 secs 232.0 MiBErrors during compilation:
        DistriMedia\Connector\Model\Order\Pdf\Invoice
                Incompatible argument type: Required type: \Magento\Store\Model\App\Emulation. Actual type: \Magento\Framework\Locale\ResolverInterface; File: 
/xxx/vendor/distrimedia/magento2-distrimedia-connector/Model/Order/Pdf/Invoice.php
Total Errors Count: 1
```

**Comment by duckchip** [comment]
Reference to issue: https://github.com/magento/devdocs/issues/8299

**Comment by duckchip** [comment]
This issue is fixed in a new release: 1.0.3
